sql: panic in execute if client supplies no data for a placeholder in bind

I am running this prepared query:
```
INSERT INTO mono ( val, sts, node, tb ) VALUES ( $1, $2, $3, $4 ) RETURNING *
```
and the client (due to a bug in my code) provide the following values down the wire:
```
    Parameters: 4
        Type OID: 20
        Type OID: 20
        Type OID: 0
        Type OID: 20
PostgreSQL
    Type: Bind
    Length: 60
    Portal: 
    Statement: 
    Parameter formats: 4
        Format: Binary (1)
        Format: Binary (1)
        Format: Text (0)
        Format: Binary (1)
    Parameter values: 4
        Column length: 8
        Data: 0000000000000000
        Column length: 8
        Data: 14364b6f35e1cd4d
        Column length: -1
        Column length: 8
```

(the 3rd placeholder is passed as "nil" (empty value))

When this is sent by the client down the wire the server fails with:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x38 pc=0xc22132]

goroutine 569 [running]:
panic(0x1592aa0, 0xc82000e150)
        /home/kena/go1.6/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/sql/parser.WalkExpr(0x7fb233326da8, 0xc82078c160, 0x7fb233326cc8, 0xc820aa0010, 0x0, 0x0, 0x1)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:387 +0x92
github.com/cockroachdb/cockroach/sql/parser.walkExprSlice(0x7fb233326da8, 0xc82078c160, 0xc8207b1300, 0x4, 0x4, 0x0, 0x0, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:292 +0xb1
github.com/cockroachdb/cockroach/sql/parser.(*Tuple).Walk(0xc82071df20, 0x7fb233326da8, 0xc82078c160, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:324 +0x64
github.com/cockroachdb/cockroach/sql/parser.WalkExpr(0x7fb233326da8, 0xc82078c160, 0x7fb233326e90, 0xc82071df20, 0x7fb233326e90, 0xc82071df20, 0x1626568)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:387 +0x98
github.com/cockroachdb/cockroach/sql/parser.(*Values).WalkStmt(0xc82071df00, 0x7fb233326da8, 0xc82078c160, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:682 +0xe9
github.com/cockroachdb/cockroach/sql/parser.WalkStmt(0x7fb233326da8, 0xc82078c160, 0x7fb233326e18, 0xc82071df00, 0x0, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:710 +0xbc
github.com/cockroachdb/cockroach/sql/parser.(*Insert).WalkStmt(0xc820666c80, 0x7fb233326da8, 0xc82078c160, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:475 +0xac
github.com/cockroachdb/cockroach/sql/parser.WalkStmt(0x7fb233326da8, 0xc82078c160, 0x7fb233326d48, 0xc820666c80, 0x0, 0x0, 0x2)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:710 +0xbc
github.com/cockroachdb/cockroach/sql/parser.FillArgs(0x7fb233326d48, 0xc820666c80, 0x7fb2333b92a8, 0xc820089330, 0x0, 0x0, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/parser/walk.go:781 +0xf3
github.com/cockroachdb/cockroach/sql.(*Executor).execStmt(0xc8200ac400, 0x7fb233326d48, 0xc820666c80, 0xc820088d00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:473 +0x501
github.com/cockroachdb/cockroach/sql.(*Executor).execStmts(0xc8200ac400, 0xc820818189, 0x4d, 0xc820088d00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:378 +0x327
github.com/cockroachdb/cockroach/sql.(*Executor).ExecuteStatements(0xc8200ac400, 0xc820815009, 0x4, 0xc820815017, 0x6, 0x0, 0xc8201ac680, 0x0, 0x0, 0x0, ...)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/executor.go:325 +0x6ec
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).executeStatements(0xc820a4ec40, 0xc820818189, 0x4d, 0xc8207b1100, 0x4, 0x4, 0xc8207effa0, 0x4, 0x4, 0x0, ...)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:555 +0xf1
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).handleExecute(0xc820a4ec40, 0xc820a4ec78, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:547 +0x329
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).serve(0xc820a4ec40, 0x0, 0x0, 0x0)
        /home/kena/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:267 +0xec3
```

--- a/sql/parser/walk.go
+++ b/sql/parser/walk.go
@@ -769,6 +769,9 @@ func (v *argVisitor) VisitPre(expr Expr) (recurse bool, newExpr Expr) {
 		v.err = fmt.Errorf("arg %s not found", placeholder)
 		return false, expr
 	}
+	if d == nil {
+		d = DNull
+	}
 	return true, d
 }
 

--- a/sql/pgwire_test.go
+++ b/sql/pgwire_test.go
@@ -359,6 +359,7 @@ func TestPGPreparedQuery(t *testing.T) {
 
 		"INSERT INTO d.T VALUES ($1) RETURNING 1": {
 			base.Params(1).Results(1),
+			base.Params(nil).Results(1),
 		},
 		/* TODO(mjibson): fix #4658
 		"INSERT INTO d.T VALUES ($1) RETURNING $1": {

